functions/__sf_section_golang.fish

```diff
@@ -0,0 +1,43 @@
+#
+# Go
+#
+# Go is an open source programming language that makes it easy
+# to build efficient software.
+# Link: https://golang.org/
+
+function __sf_section_golang -d "Display the current go version if you're inside GOPATH"
+	# ------------------------------------------------------------------------------
+	# Configuration
+	# ------------------------------------------------------------------------------
+
+	__sf_util_set_default SPACEFISH_GOLANG_SHOW true
+	__sf_util_set_default SPACEFISH_GOLANG_PREFIX $SPACEFISH_PROMPT_DEFAULT_PREFIX
+	__sf_util_set_default SPACEFISH_GOLANG_SUFFIX $SPACEFISH_PROMPT_DEFAULT_SUFFIX
+	__sf_util_set_default SPACEFISH_GOLANG_SYMBOL "🐹 "
+	__sf_util_set_default SPACEFISH_GOLANG_COLOR cyan
+
+	# ------------------------------------------------------------------------------
+	# Section
+	# ------------------------------------------------------------------------------
+
+	[ $SPACEFISH_GOLANG_SHOW = false ]; and return
+
+	if not test -d Godeps \
+		-o -f glide.yaml \
+		-o (count *.go) -gt 0 \
+		-o ([ -n $GOPATH ]; and string match $GOPATH $PWD)
+		return
+	end
+
+	if type -q go
+		set -g sf_go_version (go version | awk '{ print substr($3, 3) }')
+	else
+		return
+	end
+
+	__sf_lib_section \
+		$SPACEFISH_GOLANG_COLOR \
+		$SPACEFISH_GOLANG_PREFIX \
+		$SPACEFISH_GOLANG_SYMBOL"v"$sf_go_version \
```

Comment:
Fish has some quirky behavior when concatenating variables when one is null. If the variable isn't in quotes, the whole string concatenation will be null if any of the variables are:

![image](https://user-images.githubusercontent.com/4658208/45834651-ba08e080-bcd5-11e8-9b8c-f9923d1ef8ab.png)

So I would suggest instead writing this as `"$SPACEFISH_GOLANG_SYMBOL"v"$sf_go_version"` 👍

Reply:
Ohhhhh total fish noob mistake there lol I'm still getting used to using it. Really good to know though!

Reply:
No worries! I personally haven't found it documented anywhere.
I learned about it by being stumped by it for a few days. 😅